content/scalability/docs/workloads.md

```diff
@@ -0,0 +1,89 @@
+
+Not all controllers use EndpointSlices by default. You should verify your controller settings and enable it if needed. For the [AWS Load Balancer Controller](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/deploy/configurations/#controller-command-line-flags) you should enable the --enable-endpoint-slices optional flag to use EndpointSlices.
+
+## Use immutable and external secrets if possible
+
+The kubelet keeps a cache of the current keys and values for the Secrets that are used in volumes for pods on that node. The kubelet sets a watch on the Secrets to detect changes. As the cluster scales, the growing number of watches can negatively impact the API server performance.
+
+There are two strategies to reduce the number of watches on Secrets:
+
+* For applications that don’t need access to Kubernetes resources, you can disable auto-mounting service account secrets by setting automountServiceAccountToken: false
+* If your application’s secrets are static and will not be modified in the future, mark the [Secret as Immutable](https://kubernetes.io/docs/concepts/configuration/secret/#secret-immutable). The kubelet does not need maintain a watch for immutable secrets.
+
+To disable automatically mounting a service account to pods you can use the following setting in your workload. You can override these settings if specific workloads need a service account.
+
+```
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: app
+automountServiceAccountToken: true
+```
+
+Monitor the number of secrets in the cluster before it exceeds the limit of 10,000. You can get a total count of secrets in a cluster with the following commend.
+
+```
+kubectl get secrets -A | wc -l
+```
+
+You should set up monitoring to alert a cluster admin before this limit is reached. Consider using external secrets management options such as [AWS Key Management Service (AWS KMS)](https://aws.amazon.com/kms/) or [Hashicorp Vault](https://www.vaultproject.io/).
+
+## Limit Deployment history
+
+Pods can be slow when creating, updating, or deleting because old objects are still tracked in the cluster. You can reduce `therevisionHistoryLimit` of [deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#clean-up-policy) to cleanup older ReplicaSets which will lower to total amount of objects tracked by the Kubernetes Controller Manager. The default history limit for Deployments in 10.
+
+If your cluster creates a lot of job objects through CronJobs or other mechanisms you should use the ttlSecondsAfterFinished setting to automatically clean up old pods in the cluster.
+
+## Disable enableServiceLinks by default
```

Comment:
Should you always do this or only in certain situations? Which ones?

Reply:
In a large scale cluster I believe the answer is yes you should always disable it by default. Services can still be discovered via coredns which is a more reliable and scalable way to find service endpoints